src/main/java/org/home/model/MessageQueue.java

```diff
@@ -0,0 +1,52 @@
+package org.home.model;
+
+import java.util.LinkedList;
+import java.util.Queue;
+
+public class MessageQueue {
+    private Queue<Message> allMessage;
```

Comment:
Think about the design again. 
If you think deeply, you are letting your model to store the messages then. 
Think about the model again -> `model` can represent the how the data looks, not what it is the `data`. 
A better approach will be, give `MessageQueue` a reference or abstraction of accessing the data allocated for the queue.

Reply:
If you are to move to FileBased storage or DataBase based storage, you will have to keep everything in memory if you go by this model, then there is no point of having a File or Database